### PR TITLE
Fix flakey TestGetSetPods

### PR DESCRIPTION
**Pull request by yujuhong**

### CODE REVIEW

**Comment by yujuhong** [comment]
This fixes #5891

**Comment by vmarmol** [comment]
Oh man, rebase already :(

File: pkg/kubelet/pod_manager_test.go [hunk 1]
```diff
@@ -68,26 +68,32 @@ func TestGetSetPods(t *testing.T) {
...
+			if actual.UID == expected.UID {
+				if !reflect.DeepEqual(&expected, &actual) {
+					t.Errorf("pod was recorded incorrectly. expect: %v, got: %v", expected, actual)
+				}
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("pod %q was not found in the pod manager", expected.UID)
```
Comment (vmarmol) [inline pkg/kubelet/pod_manager_test.go]: Maybe add the pods that do exist?

**Comment by vmarmol** [comment]
LGTM

File: pkg/kubelet/pod_manager_test.go (hunk 1 above)
Comment (yujuhong) [inline pkg/kubelet/pod_manager_test.go]: Added a length comparison and dump everything if length does not match.

Comment (vmarmol) [inline pkg/kubelet/pod_manager_test.go]: Sorry, I meant in the error (in case that wasn't clear since I haven't seen your new push)

Comment (yujuhong) [inline pkg/kubelet/pod_manager_test.go]: Done.

**Comment by vmarmol** [comment]
LGTM, thanks for the quick fix @yujuhong!

**Pull request merged**